src/context/currentUser.js

```diff
@@ -0,0 +1,71 @@
+import { createContext, useEffect, useState } from 'react'
+import useAuth from '../hooks/useAuth'
+import jwt_decode from 'jwt-decode'
+import { getUser } from '../service/apiClient'
+
+export const CurrentUserContext = createContext()
+
+export const CurrentUserProvider = ({ children }) => {
+    const { token } = useAuth()
+    const [currentUser, setCurrentUser] = useState({ empty: true })
+    const [userInitials, setUserInitals] = useState('')
+    const [userFullName, setUserFullName] = useState('')
+    const [userCohort, setUserCohort] = useState('')
+    const [userFirstNameAndInital, setUserFirstNameAndInital] = useState('')
+
+    useEffect(() => {
+        async function getUserFromToken() {
+            if (token) {
+                const { userId } = jwt_decode(token)
+
+                const userDetails = await getUser(userId)
+                userDetails.status === 'success'
+                    ? setCurrentUser({ ...userDetails.data.user })
+                    : setCurrentUser({ empty: true })
+                return
+            }
+            setCurrentUser({ empty: true })
+            return
+        }
+        getUserFromToken()
+    }, [token])
+
+    useEffect(() => {
+        if (!currentUser.empty) {
+            setUserInitals(
+                currentUser
+                    ? `${currentUser.firstName[0].toUpperCase()}${currentUser.lastName[0].toUpperCase()}`
+                    : ''
+            )
+            setUserFullName(
+                currentUser
+                    ? `${currentUser.firstName} ${currentUser.lastName}`
+                    : ''
+            )
+            setUserCohort(
+                currentUser && currentUser.cohort
+                    ? 'Cohort' + currentUser.cohort
+                    : ''
+            )
+            setUserFirstNameAndInital(
+                currentUser
+                    ? `${currentUser.firstName} ${currentUser.lastName[0]}`
+                    : ''
+            )
```

Comment:
I can't let you get away with these ternarys, you fiend. Worst case, use readable if statements instead. Not sure if this is a side-effect of having this `user.empty` thing I touched on above.

Additionally, is all of this state really necessary? Why not just store the `user` object in the Auth context provider so all of this data is available and the relevant components can just use that data? You get the user object in the response that contains the token, so you'd only need to make a second fetch to get the users details if they're loading a token from localStorage and there is no user currently in state